pkg/ocm/logs.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ocm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	errors "github.com/zgalor/weberr"
+)
+
+func GetInstallLogs(client *cmv1.ClustersClient, clusterID string, tail int) (logs *cmv1.Log, err error) {
+	logsClient := client.Cluster(clusterID).Logs().Install().Log("hive")
+	response, err := logsClient.Get().
+		Parameter("tail", tail).
+		Send()
+	if err != nil {
+		err = fmt.Errorf("Failed to get logs for cluster '%s': %v", clusterID, err)
+		if response.Status() == http.StatusNotFound {
+			err = errors.NotFound.UserErrorf("Failed to get logs for cluster '%s'", clusterID)
+		}
+		return
+	}
+
+	return response.Body(), nil
+}
+
+func GetUninstallLogs(client *cmv1.ClustersClient, clusterID string, tail int) (logs *cmv1.Log, err error) {
+	logsClient := client.Cluster(clusterID).Logs().Uninstall()
+	response, err := logsClient.Get().
+		Parameter("tail", tail).
+		Send()
+	if err != nil {
+		err = fmt.Errorf("Failed to get logs for cluster '%s': %v", clusterID, err)
+		if response.Status() == http.StatusNotFound {
+			err = errors.NotFound.UserErrorf("Failed to get logs for cluster '%s'", clusterID)
+		}
+		return
+	}
+
+	return response.Body(), nil
+}
+
+func PollInstallLogs(client *cmv1.ClustersClient, clusterID string,
+	cb func(*cmv1.LogGetResponse) bool) (logs *cmv1.Log, err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
```

Comment:
What are the chances of an install taking longer than an hour? I guess you can just restart the logs.

Reply:
I figured an hour was a good compromise since the log detach is not currently working, so it will exit after an hour regardless. But I can bump this to 4 hours or something once I get the log detach correctly implemented

Reply:
:+1: Sounds good to me, I think its likely an install will take more than an hour occasionally. Probably ok for the uninstall log.